Complete set of changes to enable picolibc by default

keith-packard opened the pull request:
This series incorporates all of the patches necessary to switch to using picolibc by default. All of them are also available as separate PRs, this PR is designed to validate that the full set works as desired.

This is intended to have the changes mentioned in #49922

keith-packard committed:
libc: Default to picolibc where supported

This switches the default C library to picolibc for all targets which
support it.

Signed-off-by: Keith Packard <keithp@keithp.com>

cfriedt reviewed (commented):
LGTM - please go ahead and merge https://github.com/zephyrproject-rtos/sof/pull/28

keith-packard committed:
tests/kernel: Also run tests using minimallibc

With picolibc being the default C library, we need to explicitly include
testing against the minimal C library for kernel components.

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard committed:
tests/kernel/interrupt: Enable 'icount' for qemu_cortex_m0

This test fails when icount is disabled, so enable it

See 7bdc621ba9e8b2ab9db8a5931676b542a648802c which disabled icount by
default for this platform.

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard commented:
This PR is no longer necessary as #57340 works with no additional patches! I'm going to close it; please feel free to review #57340 if you like.
